contracts/utils/ReentrancyGuard.sol

```diff
@@ -21,8 +23,19 @@ pragma solidity ^0.8.20;
  * TIP: If you would like to learn more about reentrancy and alternative ways
  * to protect against it, check out our blog post
  * https://blog.openzeppelin.com/reentrancy-after-istanbul/[Reentrancy After Istanbul].
+ *
+ * As of v5.5, this storage-based version of the reentrancy guard is DEPRECATED. In the
+ * next major release (v6.0), {ReentrancyGuardTransient} will be removed, and its logic
+ * moved to this contract, making transient storage the standard storage option for
+ * reentrancy guards.
```

Comment:
I'd suggest adding an `IMPORTANT` prefix (or similar) for the docs. I think this is how we've historically communicated deprecations. See [Arrays#findUpperBound](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/d183d9b07a6cb0772ff52aa4e3e40165e99d6359/contracts/utils/Arrays.sol#L220-L222)

```suggestion
 * IMPORTANT: As of v5.5, this storage-based version of the reentrancy guard is DEPRECATED. In the
 * next major release (v6.0), {ReentrancyGuardTransient} will be removed, and its logic
 * moved to this contract, making transient storage the standard storage option for
 * reentrancy guards.
```

Reply:
Also clearer and shorter:

```suggestion
 * IMPORTANT: Deprecated. This storage-based reentrancy guard will be removed and replaced
 * by the {ReentrancyGuardTransient} variant in v6.0.
```